go.mod

```diff
@@ -33,6 +33,7 @@ require (
 	github.com/prometheus/client_golang v1.11.0
 	github.com/scaleway/scaleway-sdk-go v1.0.0-beta.7
 	github.com/sethvargo/go-password v0.2.0
+	github.com/terraform-providers/terraform-provider-nutanix v1.1.0
```

Comment:
This module is pointing to the outdated and archived repository, but fresh and updating https://github.com/nutanix/terraform-provider-nutanix has the same module name! which leads to to confusion and problems.

Nutanix don't have dedicated SDK module, but we can extract relevant bits from this terraform provider, using proper mod name and avoid unnecessary dependencies.

TL;DR:
Please don't import terraform provider but instead extract only relevant part (client SDK) into dedicated module.

Reply:
I agree this isn't pretty and somewhat confusing since the override to the correct location happens later on (https://github.com/kubermatic/machine-controller/pull/1132/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R63). I discussed this with @moadqassem when we started and we determined it was the best thing to do right now.

Nutanix is working on pulling out the code themselves (https://github.com/nutanix/terraform-provider-nutanix/pull/318#issuecomment-991077192), so it didn't seem sensible to do it just to undo it hopefully soon. In addition, the terraform-provider code is licensed under MPL-2.0, I'm not sure we can just copy that code into our codebase.

Reply:
`github.com/embik/nutanix-client` ;) Once nutanix have their SDK put together, we will switch.

Reply:
Done!